Neither "echo -n" nor "echo ... \c" is POSIX compliant.

See "man 1p echo" section "APPLICATION USAGE" or
https://www.in-ulm.de/%7Emascheck/various/echo+printf/

--- di-4.47/mkconfig/bin/shellfuncs.sh
+++ di-4.47/mkconfig/bin/shellfuncs.sh
@@ -45,15 +45,7 @@
 
 test_echo () {
   # It was suggested to use printf().
-  # printf does not quite work the same in the Tru64 shell.
-  # Of course no one uses that any more...
-  _tEN='-n'
-  _tEC=''
-  if [ "`echo -n test`" = "-n test" ]; then
-    _tEN=''
-    _tEC='\c'
-  fi
-  eval 'putsnonl () { echo ${_tEN} "$*"${_tEC}; }'
+  eval 'putsnonl () { printf '%s' "$*"; }'
   eval 'puts () { echo "$*"; }'
 }
 
